proposals/bring_your_own_argo_workflows/byoaw_test_plan.md

```diff
@@ -86,212 +120,547 @@ This section covers all positive functional tests to make sure that feature work
+| **Test Case Summary** | Verify run of "For loop constructs" pipeline                                                                                                                |
+| **Test Steps**        | <ol><li> Configure DSPA with external Argo </li><li> Execute Pipeline - For loop constructs</li><li> Verify each pipeline type executes correctly</li></ol> |                                                                                                                                                                                   |
+| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>DAGs inside the for loop are interated over correctly</li></ul>                                                  |
+
+
+| Test Case ID          | TC-PF-005                                                                                                                                                      |
+|-----------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Test Case Summary** | Verify run of "Parallel for execution" pipeline                                                                                                                |
+| **Test Steps**        | <ol><li> Configure DSPA with external Argo </li><li> Execute Pipeline - Parallel for execution</li><li> Verify each pipeline type executes correctly</li></ol> |                                                                                                                                                                                   |
+| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Parallel DAGs running in parallel and completes successfully</li></ul>                                              |
+
+
+| Test Case ID          | TC-PF-006                                                                                                                                                                                       |
+|-----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Test Case Summary** | Verify run of "Custom root KFP components" pipeline                                                                                                                                             |
+| **Test Steps**        | <ol><li> Configure DSPA with external Argo </li><li> Execute Pipeline - Custom root KFP components </li><li> Verify each pipeline type executes correctly </li></ol>                            |
+| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Artifcats are uploaded in the custom S3 bucket rather than the default, and downstream components are consuming from this custom location</li></ul>  |
```

Comment:
```suggestion
| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Artifacts are uploaded to the custom S3 bucket rather than the default, and downstream components are consumed from this custom location</li></ul>  |
```